packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_phone_confirmation/ui_components/phone_number_form.dart

```diff
@@ -5,105 +5,92 @@ import 'package:provider/provider.dart';
-              SizedBox(width: context.designSystem.spacing.xs),
-              Expanded(child: _buildSelectCountry(context)),
-            ],
-          ),
-          SizedBox(height: context.designSystem.spacing.m),
-          Row(
-            crossAxisAlignment: CrossAxisAlignment.center,
-            children: [
-              Icon(
-                Icons.phone,
-                color: context.designSystem.colors.gray,
-                size: context.designSystem.spacing.xxl,
-              ),
-              SizedBox(width: context.designSystem.spacing.xs),
-              Flexible(
-                child: _buildPhoneNumber(context, hint: phoneNumberHint),
-              ),
-            ],
-          ),
-        ],
-      );
+  State<PhoneNumberForm> createState() => _PhoneNumberFormState();
+}
 
-  /// region Builders
+class _PhoneNumberFormState extends State<PhoneNumberForm> {
+  final _phoneNumberFocusNode = FocusNode(debugLabel: 'phoneNumberFocus');
 
-  Widget _buildSelectCountry(BuildContext context) =>
-      RxBlocBuilder<OnboardingPhoneBlocType, CountryCodeModel?>(
-        state: (bloc) => bloc.states.countryCode,
-        builder: (context, countryCodeSnapshot, bloc) => PrimaryButton(
-          onPressed: () => _onCountryCodeSegmentPressed(
-            context,
-            countryCodeSnapshot.data,
-          ),
-          child: Padding(
-            padding: EdgeInsets.symmetric(
-              vertical: context.designSystem.spacing.m,
-            ),
-            child: Text(
-              countryCodeSnapshot.data?.itemDisplayName ??
-                  context.l10n.featureOnboarding.countrySelectionLabel,
-              textAlign: TextAlign.center,
-              style: context.designSystem.typography.h2Reg16,
-            ),
-          ),
-        ),
-      );
+  @override
+  void initState() {
+    super.initState();
+    _phoneNumberFocusNode.requestFocus();
+  }
 
-  Widget _buildPhoneNumber(
-    BuildContext context, {
-    String hint = 'XX XXX XXXX',
-  }) =>
+  @override
+  void dispose() {
+    FocusManager.instance.primaryFocus?.unfocus();
+    _phoneNumberFocusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) =>
       RxTextFormFieldBuilder<OnboardingPhoneBlocType>(
         state: (bloc) => bloc.states.phoneNumber.translate(context),
         onChanged: (bloc, value) => bloc.events.setPhoneNumber(value),
         showErrorState: (bloc) => bloc.states.showErrors,
         builder: (fieldState) => TextFormField(
-          textAlign: TextAlign.center,
+          focusNode: _phoneNumberFocusNode,
           keyboardType: TextInputType.phone,
           controller: fieldState.controller,
           onEditingComplete: () =>
               FocusManager.instance.primaryFocus?.unfocus(),
           decoration: fieldState.decoration.copyWith(
-            hintText: hint,
-            errorText: fieldState.error,
+            prefixIcon: _buildSelectCountry(context),
```

Comment:
Please confirm the `prefix` property is not suitable, that's why the `prefixIcon` is used instead

Reply:
The difference between `prefix` and `prefixIcon` parameters is minimal, with the exception of order of items displayed if specified. The `prefixIcon` is displayed before the prefix and within the decoration's container. The choice of the `prefixIcon` is arbitrary, but can be changed if needed.